Index: ext/ogg/gstoggdemux.c
===================================================================
RCS file: /cvs/gstreamer/gst-plugins-base/ext/ogg/gstoggdemux.c,v
retrieving revision 1.195
diff -u -r1.195 gstoggdemux.c
--- ext/ogg/gstoggdemux.c	7 Aug 2008 15:58:50 -0000	1.195
+++ ext/ogg/gstoggdemux.c	9 Aug 2008 11:55:54 -0000
@@ -1746,6 +1746,9 @@
   for (i = 0; i < chain->streams->len; i++) {
     GstOggPad *pad = g_array_index (chain->streams, GstOggPad *, i);
 
+    if (pad->is_skeleton)
+      continue;
+
     gst_pad_push_event (GST_PAD_CAST (pad), gst_event_new_eos ());
 
     GST_DEBUG_OBJECT (ogg, "removing pad %" GST_PTR_FORMAT, pad);
@@ -1786,6 +1789,10 @@
       GstOggPad *pad;
 
       pad = g_array_index (chain->streams, GstOggPad *, i);
+
+      if (pad->is_skeleton)
+        continue;
+
       GST_DEBUG_OBJECT (ogg, "adding pad %" GST_PTR_FORMAT, pad);
 
       /* mark discont */
